src/modules/roc_pipeline/receiver_session.cpp

```diff
@@ -126,6 +126,14 @@ ReceiverSession::ReceiverSession(const SessionConfig& config,
     }
 #endif // ROC_TARGET_OPENFEC
 
+    if (format->sample_rate != out_sample_rate && !config.resampling) {
+        roc_log(LogError,
+                "receiver_session: input and output sample rates must be equal"
```

Comment:
This is quite a low level check. ReceiverSession is a high level component that just brings other components together, so I think we should move this check somewhere else.

Actually, we already have a place where we check input and output rates when resampler is enabled: https://github.com/roc-project/roc/blob/develop/src/modules/roc_audio/latency_monitor.cpp#L48, so i think we should move this check to LatencyMonitor as well.

Reply:
Done.